maas/resource_maas_network_interface_tag_test.go

```diff
@@ -0,0 +1,138 @@
+package maas_test
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"terraform-provider-maas/maas"
+	"terraform-provider-maas/maas/testutils"
+)
+
+func TestSplitTagStateId(t *testing.T) {
+	expectedSystemId := "acb123"
+	expectedInterfaceId := 12
+	stateId := fmt.Sprintf("%s:%d", expectedSystemId, expectedInterfaceId)
+	systemId, interfaceId, err := maas.SplitTagStateId(stateId)
+	if err != nil {
+		t.Fatalf("Error splitting state ID: %s", err)
+	}
+	if systemId != expectedSystemId || interfaceId != expectedInterfaceId {
+		t.Fatalf("Expected system ID %s and interface ID %d, got system ID %s and interface ID %d", expectedSystemId, expectedInterfaceId, systemId, interfaceId)
+	}
+}
+
+func TestAccNetworkInterfaceTag_basic(t *testing.T) {
+	hostname := acctest.RandomWithPrefix("tf")
+	macAddress := testutils.RandomMAC()
+	tagName := acctest.RandomWithPrefix("tag")
+	tagName2 := acctest.RandomWithPrefix("tag2")
+	tagName3 := acctest.RandomWithPrefix("tag3")
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testutils.PreCheck(t, nil) },
+		Providers:    testutils.TestAccProviders,
+		ErrorCheck:   func(err error) error { return err },
+		CheckDestroy: testAccCheckMaasNetworkInterfaceDestroy,
+		Steps: []resource.TestStep{
+			// Test creation.
+			{
+				Config: testAccMaasNetworkInterfaceTagConfig(hostname, macAddress, tagName, tagName2),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckMaasNetworkInterfaceTagExists("maas_network_interface_tag.test"),
+					resource.TestCheckResourceAttr("maas_network_interface_tag.test", "tags.#", "2"),
+					resource.TestCheckTypeSetElemAttr("maas_network_interface_tag.test", "tags.*", tagName),
+					resource.TestCheckTypeSetElemAttr("maas_network_interface_tag.test", "tags.*", tagName2),
+				),
+			},
+			// Test update. Expected behaviour is that the previous tag is removed and the new tag is added.
+			{
+				Config: testAccMaasNetworkInterfaceTagConfig(hostname, macAddress, tagName2, tagName3),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckMaasNetworkInterfaceTagExists("maas_network_interface_tag.test"),
+					resource.TestCheckResourceAttr("maas_network_interface_tag.test", "tags.#", "2"),
+					resource.TestCheckTypeSetElemAttr("maas_network_interface_tag.test", "tags.*", tagName2),
+					resource.TestCheckTypeSetElemAttr("maas_network_interface_tag.test", "tags.*", tagName3),
+				),
+			},
+			// Test import.
```

Comment:
How is this test going to work? I would expect we specify to Terraform how to calculate the ID with `ImportStateIdFunc` like we do in couple other tests